src/ring_theory/polynomial/basic.lean

```diff
@@ -758,6 +758,128 @@ begin
   simp only [monomial_eq, ϕ.map_pow, ϕ.map_prod, ϕ.comp_apply, ϕ.map_mul, finsupp.prod_pow],
 end
 
+lemma quotient_map_C_eq_zero {I : ideal R} :
+  ∀ i ∈ I, (ideal.quotient.mk (ideal.map C I : ideal (mv_polynomial σ R))).comp C i = 0 :=
+begin
+  intros i hi,
+  simp only [function.comp_app, ring_hom.coe_comp, ideal.quotient.eq_zero_iff_mem],
+  exact ideal.mem_map_of_mem hi
+end
+
+/-- If every coefficient of a polynomial is in an ideal `I`, then so is the polynomial itself,
+multivariate version. -/
+lemma mem_ideal_of_coeff_mem_ideal (I : ideal (mv_polynomial σ R)) (p : mv_polynomial σ R)
+  (hcoe : ∀ (m : σ →₀ ℕ), (p.coeff m) ∈ I.comap C) : p ∈ I :=
+begin
+  rw as_sum p,
+  suffices : ∀ m ∈ p.support, monomial m (mv_polynomial.coeff m p) ∈ I,
+  { exact submodule.sum_mem I this },
+  intros m hm,
+  rw [← mul_one (coeff m p), ← C_mul_monomial],
+  suffices : C (coeff m p) ∈ I,
+  { exact ideal.mul_mem_right I (monomial m 1) this },
+  simpa [ideal.mem_comap] using hcoe m
+end
+
+/-- The push-forward of an ideal `I` of `R` to `mv_polynomial σ R` via inclusion
+ is exactly the set of polynomials whose coefficients are in `I` -/
+theorem mem_map_C_iff {I : ideal R} {f : mv_polynomial σ R} :
+  f ∈ (ideal.map C I : ideal (mv_polynomial σ R)) ↔ ∀ (m : σ →₀ ℕ), f.coeff m ∈ I :=
+begin
+  split,
+  { intros hf,
+    apply submodule.span_induction hf,
+    { intros f hf n,
+      cases (set.mem_image _ _ _).mp hf with x hx,
+      rw [← hx.right, coeff_C],
+      by_cases (n = 0),
+      { simpa [h] using hx.left },
+      { simp [ne.symm h] } },
+    { simp },
+   { exact λ f g hf hg n, by simp [I.add_mem (hf n) (hg n)] },
+    { refine λ f g hg n, _,
+      rw [smul_eq_mul, coeff_mul],
+      exact I.sum_mem (λ c hc, I.smul_mem (f.coeff c.fst) (hg c.snd)) } },
+  { intros hf,
+    rw as_sum f,
+    suffices : ∀ m ∈ f.support, monomial m (coeff m f) ∈
+      (ideal.map C I : ideal (mv_polynomial σ R)),
+    { exact submodule.sum_mem _ this },
+    intros m hm,
+    rw [← mul_one (coeff m f), ← C_mul_monomial],
+    suffices : C (coeff m f) ∈ (ideal.map C I : ideal (mv_polynomial σ R)),
+    { exact ideal.mul_mem_right _ _ this },
+    apply ideal.mem_map_of_mem _,
+    exact hf m }
+end
+
+lemma eval₂_C_mk_eq_zero {I : ideal R} :
+  ∀ f ∈ (ideal.map C I : ideal (mv_polynomial σ R)),
+  eval₂_hom (C.comp (ideal.quotient.mk I)) X f = 0 :=
+begin
+  intros a ha,
```

Comment:
```suggestion
lemma eval₂_C_mk_eq_zero {I : ideal R} {a} (ha : a ∈ (ideal.map C I : ideal (mv_polynomial σ R)) :
  eval₂_hom (C.comp (ideal.quotient.mk I)) X a = 0 :=
begin
```
or with `f` instead of `a, but then you'll need to change the below